### PR TITLE
Horizontal space at value relation widget at multiple selection

### PR DESCRIPTION
**Pull request by signedav**
## Description
When in the value relation widget allow multiple selection is active, the widget was displayed as QListWidget where you could select the related items. This could have been not very nice when there were a lot of items. It's now a QTableWidged with multiple columns. It's much more comfortable to handle a lot of items and in case there are very few items it can be made very small and we can save some space. The number of columns is configurable.

### Now
multiselect_is
(if only one column it looks the same like before)

### Before
multiselect_was

The listWidget parts are completely replaced by the tableWidget (including tests) and removed from qgsvaluerelationsearchwidgetwrapper where it didn't made sense anyway.

**This is a second pull request to the same improvements like #5954 - what is wrecked because of wrong merging stuff...**

### CODE REVIEW

File: src/gui/editorwidgets/qgsvaluerelationwidgetwrapper.h [hunk 1]
```diff
@@ -20,7 +20,8 @@
 #include "qgsvaluerelationfieldformatter.h"
 
 #include <QComboBox>
-#include <QListWidget>
+#include <QTableWidget>
+#include <QHeaderView>
```
Comment (m-kuhn) [inline src/gui/editorwidgets/qgsvaluerelationwidgetwrapper.h]: I think these includes should go to the cpp file (and a forward declaration in here).

File: src/gui/editorwidgets/qgsvaluerelationwidgetwrapper.cpp [hunk 2]
```diff
@@ -23,6 +23,7 @@
 #include "qgsfilterlineedit.h"
 #include "qgsfeatureiterator.h"
 #include "qgsvaluerelationfieldformatter.h"
+#include "qgslogger.h"
```
Comment (m-kuhn) [inline src/gui/editorwidgets/qgsvaluerelationwidgetwrapper.cpp]: Is this include required?

**Review by m-kuhn — COMMENTED** [review]
Looks mostly good to me. Just two minor comments regarding includes and then it's ready to merge. Thanks a lot.

File: src/gui/editorwidgets/qgsvaluerelationwidgetwrapper.cpp (hunk 2 above)
Comment (signedav) [reply src/gui/editorwidgets/qgsvaluerelationwidgetwrapper.cpp]: no

File: src/gui/editorwidgets/qgsvaluerelationwidgetwrapper.h (hunk 1 above)
Comment (signedav) [reply src/gui/editorwidgets/qgsvaluerelationwidgetwrapper.h]: ok

**Comment by m-kuhn** [comment]
Thanks Dave :+1:

**Pull request merged**